### PR TITLE
Implement service credentials API

### PR DESCRIPTION
**Pull request by ashkarpetin**
## Summary

Combines all credentials hashes resources into a service credentials API:
-  smtp
- webhook
- mailgun
- sendgrid
- aws-ses
- oauth2
- postmark
- experian
- taxjar
- avalara
- plaid

Alternative to https://github.com/Rebilly/api-definitions/pull/1169

## Links
Internal discussion

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: plugins/products-bundler/mapping/Users.yaml [hunk 1]
```diff
@@ -120,6 +120,7 @@ x-tagGroups:
       - Websites
       - Custom domains
       - Integrations
+      - Credential hashes
```
Comment (adamaltman) [inline plugins/products-bundler/mapping/Users.yaml:123]: Maybe we should rename it to `Service credentials` too?

Comment (ashkarpetin) [reply plugins/products-bundler/mapping/Users.yaml:123]: Yeah, we should, I was not sure if `Service credentials` is a good term for that

File: openapi/components/parameters/hash.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: hash
 in: path
-description: ID of the token.
+description: Hash of the credential.
```
Comment (adamaltman) [inline openapi/components/parameters/hash.yaml:3]: Is it salted? It should be, right?

Comment (ashkarpetin) [reply openapi/components/parameters/hash.yaml:3]: It is not actually a hash but an UUID. I've updated schemas and replaced it with ID. We will have to rename `credentialHash` to `serviceCredentialId` for all endpoints that accept it.

File: openapi/components/parameters/credentialType.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+name: type
+in: path
+description: Service credential type.
```
[the remark this answers is not in the document]
Comment (ashkarpetin) [reply openapi/components/parameters/credentialType.yaml]: Thank for suggestions. I've renamed the parameter.

[the remark this answers is not in the document]
Comment (ashkarpetin) [reply openapi/components/parameters/credentialType.yaml]: I have to rollback to old parameter name, used you suggestion.

File: openapi/components/parameters/hash.yaml (hunk 2 above)
Comment (adamaltman) [reply openapi/components/parameters/hash.yaml:3]: Description should say "ID of the service credential"?

Comment (ashkarpetin) [reply openapi/components/parameters/hash.yaml]: `hash` parameter is used by legacy endpoints, I updated description to `ID of the credential`. For service credentials I used `resourceId` parameter.

**Comment by ashkarpetin** [comment]
php please review

**Comment by ashkarpetin** [comment]
devs please review

**Review by jvc9109 — APPROVED** [review]

**Review by noon-dawg — APPROVED** [review]

**Review by ninjasimon — APPROVED** [review]

**Review by swilczynski — APPROVED** [review]

**Pull request merged**